codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolVisitor.kt

```diff
@@ -281,11 +313,16 @@ class SymbolVisitor(
     override fun memberShape(shape: MemberShape): Symbol {
         val target = model.expectShape(shape.target)
         val targetSymbol = this.toSymbol(target)
+        if (config.handleOptionality && config.handleRequired) {
+            throw CodegenException("CodegenVisitor 'handleOptionality' and 'handleRequired' configuration options cannot be both true at the same time")
+        }
         // Handle boxing first so we end up with Option<Box<_>>, not Box<Option<_>>
         return targetSymbol.letIf(config.handleRustBoxing) {
             handleRustBoxing(it, shape)
         }.letIf(config.handleOptionality) {
             handleOptionality(it, shape)
+        }.letIf(config.handleRequired) {
+            handleRequiredTrait(it, shape)
```

Comment:
could merge these and handle optionality in one place based on the flags

Reply:
I think I prefer to have the full logical division to be honest, but I don't have a very strong reasoning behind it.